### PR TITLE
Fwmt 1379 implement redis for gateway ccs messages

### PR DESCRIPTION
**Pull request by SteveScorfield**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -35,12 +35,61 @@ springBoot {
 	}
 }
 
+configurations {
+	jaxb
+}
+
+task genJaxb {
```
Comment (JacobHarrisonQA) [inline build.gradle]: your formatter a bit off here?

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/config/RedisConfig.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  private String redisHostname;
+  private int redisPort;
+
+  public RedisConfig(@Value("${redis.host}") String redisHostname,
+      @Value("${redis.port}") int redisPort) {
+    this.redisHostname = "localhost";
+    this.redisPort = redisPort;
+  }
+
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/config/RedisConfig.java]: do you need to connect to a certain database here ?

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/controller/OutcomeApi.java [hunk 3]
```diff
@@ -35,7 +38,7 @@
       produces = {"application/json"},
       method = RequestMethod.POST)
   ResponseEntity<CCSPropertyListingOutcome> ccsPropertyListingCaseOutcomeResponse(
-      @RequestBody CCSPropertyListingOutcome ccsPropertyListingOutcome) throws GatewayException;
+      @RequestBody CCSPropertyListingOutcome ccsPropertyListingOutcome) throws GatewayException, JsonProcessingException;
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/controller/OutcomeApi.java]: catch the JsonProcessingException and only throw GatewayException

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/controller/OutcomeController.java [hunk 4]
```diff
@@ -39,7 +47,11 @@
   }
 
   @Override
-  public ResponseEntity<CCSPropertyListingOutcome> ccsPropertyListingCaseOutcomeResponse(CCSPropertyListingOutcome ccsPLOutcome) throws GatewayException {
+  public ResponseEntity<CCSPropertyListingOutcome> ccsPropertyListingCaseOutcomeResponse(
+      CCSPropertyListingOutcome ccsPLOutcome) throws GatewayException, JsonProcessingException {
+
```
Comment (JacobHarrisonQA) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/controller/OutcomeController.java]: only throw GatewayException

File: build.gradle (hunk 1 above)
Comment (SteveScorfield) [reply build.gradle]: Reformatted

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/config/RedisConfig.java (hunk 2 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/config/RedisConfig.java]: That's done within the jedisConnectionFactory method of this class.

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/controller/OutcomeController.java (hunk 4 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/controller/OutcomeController.java]: Changed

File: build.gradle [hunk 5]
```diff
@@ -35,12 +35,61 @@ springBoot {
...
+	}
+}
 dependencies {
 	compile('uk.gov.ons.census.fwmt:census-fwmt-canonical:1.0.11-RELEASE')
 	compile('uk.gov.ons.census.fwmt:census-fwmt-events:1.0.13-RELEASE')
-	compile('uk.gov.ons.census.fwmt:census-fwmt-common:1.0.25-RELEASE')
+	compile('uk.gov.ons.census.fwmt:census-fwmt-common:1.0.29-RELEASE')
 	compile('uk.gov.ons.ctp.integration.common:product-reference:0.0.12')
 
+	jaxb("javax.xml.bind:jaxb-api:2.3.0")
```
Comment (spdiaz1000) [inline build.gradle]: Why is xml binding needed, I cant see anything in the code that needs the biinding stuff

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/JobCacheManagerImpl.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.census.fwmt.outcomeservice.config.RedisUtil;
+import uk.gov.ons.census.fwmt.outcomeservice.service.JobCacheManager;
+
+
+@Slf4j
+@Service
+public class JobCacheManagerImpl implements JobCacheManager {
```
Comment (spdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/JobCacheManagerImpl.java]: I think stick this bit of functionality in CCSPLStore, and remove JobCacheManager & Imple

File: build.gradle (hunk 5 above)
Comment (SteveScorfield) [reply build.gradle]: Not quite sure on this as I can't find it within the build.gradle

File: src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/JobCacheManagerImpl.java (hunk 6 above)
Comment (SteveScorfield) [reply src/main/java/uk/gov/ons/census/fwmt/outcomeservice/service/impl/JobCacheManagerImpl.java]: This has been moved the CCSPLStore and the related job cache classes have been removed.

File: build.gradle (hunk 5 above)
Comment (SteveScorfield) [reply build.gradle]: Removed JAXB

**Pull request merged**